helpers/run-app.ts

```diff
@@ -26,63 +26,80 @@ const createProcess = (
+  template: string,
   frontend: boolean,
   framework: TemplateFramework,
   port?: number,
   externalPort?: number,
 ): Promise<any> {
-  let backendAppProcess: ChildProcess;
-  let frontendAppProcess: ChildProcess | undefined;
-  const frontendPort = port || 3000;
-  let backendPort = externalPort || 8000;
-
-  // Callback to kill app processes
-  process.on("exit", () => {
-    console.log("Killing app processes...");
-    backendAppProcess.kill();
-    frontendAppProcess?.kill();
-  });
-
-  let backendCommand = "";
-  let backendArgs: string[];
-  if (framework === "fastapi") {
-    backendCommand = "poetry";
-    backendArgs = [
-      "run",
-      "uvicorn",
-      "main:app",
-      "--host=0.0.0.0",
-      "--port=" + backendPort,
-    ];
-  } else if (framework === "nextjs") {
-    backendCommand = "npm";
-    backendArgs = ["run", "dev"];
-    backendPort = frontendPort;
-  } else {
-    backendCommand = "npm";
-    backendArgs = ["run", "dev"];
-  }
-
-  if (frontend) {
+  if (template === "extractor") {
+    const commandArgs = ["run", "reflex", "run"];
+    if (port) {
+      commandArgs.push("--frontend-port", port.toString());
+    }
+    if (externalPort) {
+      commandArgs.push("--backend-port", externalPort.toString());
+    }
     return new Promise((resolve, reject) => {
-      backendAppProcess = createProcess(backendCommand, backendArgs, {
+      createProcess("poetry", commandArgs, {
         stdio: "inherit",
-        cwd: path.join(appPath, "backend"),
-        env: { ...process.env, PORT: `${backendPort}` },
-      });
-      frontendAppProcess = createProcess("npm", ["run", "dev"], {
-        stdio: "inherit",
-        cwd: path.join(appPath, "frontend"),
-        env: { ...process.env, PORT: `${frontendPort}` },
+        cwd: path.join(appPath),
       });
     });
   } else {
-    return new Promise((resolve, reject) => {
-      backendAppProcess = createProcess(backendCommand, backendArgs, {
-        stdio: "inherit",
-        cwd: path.join(appPath),
-        env: { ...process.env, PORT: `${backendPort}` },
-      });
+    let backendAppProcess: ChildProcess;
+    let frontendAppProcess: ChildProcess | undefined;
+    const frontendPort = port || 3000;
+    let backendPort = externalPort || 8000;
+
+    // Callback to kill app processes
+    process.on("exit", () => {
+      console.log("Killing app processes...");
+      backendAppProcess.kill();
+      frontendAppProcess?.kill();
     });
+
+    let backendCommand = "";
+    let backendArgs: string[];
+    if (framework === "fastapi") {
+      backendCommand = "poetry";
+      backendArgs = [
+        "run",
+        "uvicorn",
+        "main:app",
+        "--host=0.0.0.0",
+        "--port=" + backendPort,
+      ];
+    } else if (framework === "nextjs") {
+      backendCommand = "npm";
+      backendArgs = ["run", "dev"];
+      backendPort = frontendPort;
+    } else {
+      backendCommand = "npm";
+      backendArgs = ["run", "dev"];
+    }
```

Comment:
also i wonder why we still need this code. isn't the app automatically started by using the `runApp` flag?
seems to me like we can delete most of this

Reply:
Isn't this runApp code?

Reply:
Ah sorry. Was thinking this is the helper code from the e2b tests.

Reply:
But still looks quite complex. Any ideas how to simplify?